fastfield_codecs/src/serialize.rs

```diff
@@ -98,6 +98,29 @@ impl Header {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct U128Header {
+    pub num_vals: u32,
+    pub codec_type: U128FastFieldCodecType,
```

Comment:
I'm a  bit confused... Why do we need `num_vals` and are you planning to handle sparsity in a non-generic way?

Reply:
sparsity would be a general solution with a separate index